draft-specifications/Model.md

```diff
@@ -0,0 +1,411 @@
+> record that we agreed that strings can only be well-formed Unicode, i.e.,
+> cannot contain unpaired surrogates.
+> For JavaScript, if a string does not pass [the `isWellFormed`
+> predicate](https://github.com/tc39/proposal-is-usv-string), then it is not a
+> Passable string.
+>
+> Agoric has agreed to tentatively omit lone surrogates, pending vetting of the
+> performance of the implementation in their CapTP.
+> https://github.com/ocapn/ocapn/issues/47
+>
+> Agoric previously held the position that all valid strings that JavaScript
+> can parse with `JSON.parse` should round-trip.
+> https://github.com/ocapn/ocapn/issues/5#issuecomment-1550370096,
+> https://github.com/ocapn/ocapn/issues/5#issuecomment-1550481111
+>
+> Spritely has consistently preferred a representation that can be captured
+> with UTF-8, which cannot transit lone/unpaired surrogates.
+> (October 10, 2023)
+
+## ByteArray
+
+An array of bytes.
+
+> - **Guile**: `#vu8()`
+> - **JavaScript**: `new ArrayBuffer()`
+> - **Python**: `b''`
+>
+> Tracking: https://github.com/ocapn/ocapn/issues/48
+>
+> At the [November 14, 2023](https://github.com/ocapn/ocapn/issues/94) meeting,
+> we agreed to settle on the prefix "byte" and agreed to decide at the next
+> meeting based on a [reactji
+> poll](https://github.com/ocapn/ocapn/issues/48#issuecomment-1811097196).
+>
+> [January 2024 meeting
+> notes](https://github.com/ocapn/ocapn/blob/main/meeting-minutes/2024-01-09.md)
+> record that we agreed on ByteArray because it was the winner of the poll, and
+> we had already agreed to resolve this specific issue by poll among these
+> three choices.
+
+## Selector
+
+A string of Unicode code points excluding surrogates (U+D800-U+DFFF).
+Selectors are distinguished from [String](#string)s by type, not content.
+
+> - **Guile**: symbols `'name`
+> - **JavaScript**:
+>   ```js
+>   {
+>     [Symbol.for('passStyle'): 'selector',
+>     [Symbol.toStringTag]: 'name',
+>   }
+>   ```
+> - **Python**: `Selector('name')` where `Selector` is imported from `ocapn`.
+>
+> A selector's content must be expressible in UTF-8.
+> Strings in UTF-16 can express unpaired/lone surrogates that cannot be
+> expressed in any UTF and so cannot be carried by OCapN.
+>
+> Tracking: https://github.com/ocapn/ocapn/issues/46
+>
+> OCapN uses the name selector to avoid the implication that they will
+> correspond to a language's symbol type in all languages that have a symbol
+> type.
+> Selectors may correspond to symbols in languages where a symbol is elligible
+> for garbage collection when there are no extant references.
+> At this time, JavaScript cannot safely use registered symbols like
+> `Symbol.for('name')` for OCapN selectors, because registered symbols are
+> interned without possibility of eventual garbage collection.
+>
+> OCapN supports one operator for delivering both function application and
+> method invocation.
+> By convention, method invocation is equivalent to function application, where
```

Comment:
is the method selector (first argument) omitted in the case of function invocation?

in Spritely the method selector (first argument) is #f in function invocation, in E __run was used and in Monte it was run.

Thoughts?

Reply:
> in Spritely the method selector (first argument) is #f in function invocation

If that's true, then we have badly misunderstood how these concepts map to spritely, to a degree that does likely invalidate what we thought we agreed to.

@cwebber , @tsyesika , please clarify. Thanks.

Reply:
It’s not true. When invoking it as a function you simply provide the function arguments as the arguments. No need to pass false or do anything else.

Reply:
Pardon, this is stale. I’ll update.

Reply:
I thought it was so on the 'wire' of a Spritely captp connection.

Reply:
> I thought it was so on the 'wire' of a Spritely captp connection.

There used to be a dedicated method field on desc:deliver which was set to false on function invocation but it was never really used iirc and was dropped a long time ago

Reply:
s/desc:deliver/op:deliver/ no?

Reply:
Okay, now seated and looking at what I’ve written, the proposed specification does not suggest that non-method invocation must have False as the first argument. The text is consistent with consensus. I do recall the former idea about `#f`, but it is not current.